### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by marest94**
Added table of contents, dependency table, added new information about message handlers, etc.

Take a look at it, and if you have anything to propose, what's missing, or something similar,  I'm more than welcome to listen.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -220,9 +316,9 @@ curl --location --request POST 'https://localhost:8083/proxy' \
     "requestedArtifact" : "test1.csv"
```
Comment (IgorBalog-Eng) [inline README.md:316]: Is this one just test1.csv or it needs to be in URI format?

File: README.md [hunk 2]
```diff
@@ -76,40 +132,67 @@ This class is the entry point of [proxy request](#proxyendpoint). Business logic
...
+* **handleWssResponse(MultipartMessage mm) (WSS Flow)**
 
-### Provider side modification
+
+### Provider side modification <a name="providersidemodification"></a>
 
 For making modification when dataApp is in provider role, one of the following controllers can be used as starting point
 
-**it.eng.idsa.dataapp.web.rest.DataControllerBodyBinary**
+* **it.eng.idsa.dataapp.web.rest.DataControllerBodyBinary (REST Flow)**
```
[the remark this answers is not in the document]
Comment (marest94) [reply README.md:145]: Yes we can!

File: README.md [hunk 3]
```diff
@@ -172,7 +257,18 @@ application.ecc.wss-port=8098
 ```
 Use application.ecc.wss-port= property to set the ECC Sender WSS port.
 
-In ECC property file to use Web Socket between ECC:
+In config.properties file
+
+```
+server.ssl.key-store=ssl-server.jks
```
[the remark this answers is not in the document]
Comment (marest94) [reply README.md:263]: Done!

File: README.md (hunk 1 above)
Comment (marest94) [reply README.md:316]: You're right, changed!

**Review by IgorBalog-Eng — APPROVED** [review]
Approved

**Pull request merged**